src/udev/udev-dump.c

```diff
@@ -29,7 +30,272 @@ void event_cache_written_sysctl(UdevEvent *event, const char *attr, const char *
         event_cache_written_value(&event->written_sysctls, attr, value);
 }
 
-void dump_event(UdevEvent *event, FILE *f) {
+static int dump_event_json(UdevEvent *event, sd_json_format_flags_t flags, FILE *f) {
+        sd_device *dev = ASSERT_PTR(ASSERT_PTR(event)->dev);
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        const char *str;
+        int r;
+
+        if (sd_device_get_devpath(dev, &str) >= 0) {
+                r = sd_json_variant_set_field_string(&v, "path", str);
+                if (r < 0)
+                        return r;
+        }
+
+        if (sd_device_get_sysname(dev, &str) >= 0) {
+                r = sd_json_variant_set_field_string(&v, "name", str);
+                if (r < 0)
+                        return r;
+        }
+
+        if (sd_device_get_sysnum(dev, &str) >= 0) {
+                r = sd_json_variant_set_field_string(&v, "number", str);
```

Comment:
btw, why is this a string in the first place? at least if we output this in json, shuldn't we convert it to an integer?

Reply:
Changed to unsigned.